labs/a1.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Advanced Lab 1
+layout: lab
+---
+
+Lecture: Thurs 2/1
+Released: Mon 2/5
+Due: End of Feb
+
+# Advanced section, Lab 1
+
+## Grading note
+Labs are graded on completion. Treat this lab as seeds of exploration instead of just a grade. If you don't pass on the first submission, you can have it checked off in-person by a decal facilitator.
+
+Since you know how to use unix tools (though you may be more or less familiar with certain tools), the goal of this lab is to drop you in the wilderness. You can find your way out! :D
+
+
+## Composability & workflows
+This lab can be done on your own UNIX-like machine, or you can ssh into `tsunami.ocf.berkeley.edu` using your OCF account to finish the lab there. As always, `man` and Google will be your friends.
+
+
+### Shell. web fetching, parsing, and frequency analysis.
+- [ ] Download Alice in Wonderland by parsing out a download link from Project Gutenberg.
+
+    https://www.gutenberg.org/ebooks/11
+
+As it turns out, Project Gutenberg doesn't like users of curl, and will demand you enable javascript.  It's probably a resonable defense against naive abuse of curl by people scraping the site, but it's annoying to deal with.
+
+
+You can just tell the server that you're Firefox instead:
+
+    curl -s -A "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36"  https://www.gutenberg.org/
+    
+Credit to: [gurditsbedi on medium](https://medium.com/@gurditsbedi/automating-download-from-project-gutenberg-using-the-linux-terminal-e9f35bbbd06c)
+
+
+
+- [ ] Isolate the body content from the headers and footers, perhaps saving to an intermediate file. 
+
+Hint: 
+`less` may be useful for inspection, and`sed` with no-echo option. 
+
+`sed`'s "#,#p" line-number/print argument (play with ranges, offset, and start/end of file chars...twist on regex line terminators).
```

Comment:
This sentence feels incomplete, I'd flesh it out into a complete sentence with a little more direction